packages/theia-integration/src/browser/diagram/glsp-saveable.ts

```diff
@@ -64,6 +71,38 @@ export class GLSPSaveable implements Saveable, Disposable {
         return this.editorContextService.isDirty;
     }
 
+    // Needs to be implemented to pass the type check of `WorkspaceFrontendContribution.canBeSaved`.
+    // GLSP handles saving and model changes on the server side. The `revert` implementation is no-op and has no effect.
+    async revert(options?: Saveable.RevertOptions): Promise<void> {}
+
+    // Needs to be implemented to pass the type check of `WorkspaceFrontendContribution.canBeSaved`.
+    // GLSP only supports server-side saving. The `createSnapshot` implementation is no-op and has no effect.
+    createSnapshot(): Saveable.Snapshot {
```

Comment:
Do you remember why this was needed? I don't think we should implement this as this will enable 'Save As' and leads to an exception `Could not apply changes to new file`. I think we should simply not implement that method. Same for 'revert' unless there is a very good reason.